src/Mvc/Mvc.Core/src/Resources.resx

```diff
@@ -519,6 +519,18 @@
   <data name="ValidationVisitor_ContainerCannotBeSpecified" xml:space="preserve">
     <value>A container cannot be specified when the ModelMetada is of kind '{0}'.</value>
   </data>
+  <data name="ComplexObjectModelBinder_NoSuitableConstructor_ForParameter" xml:space="preserve">
+    <value>Could not create an instance of type '{0}'. Model bound complex types must not be abstract or value types and must have a parameterless constructor. Record types must have a single primary constructor. Alternatively, give the '{1}' parameter a non-null default value.</value>
+  </data>
+  <data name="ComplexObjectModelBinder_NoSuitableConstructor_ForProperty" xml:space="preserve">
+    <value>Could not create an instance of type '{0}'. Model bound complex types must not be abstract or value types and must have a parameterless constructor. Record types must have a single primary constructor. Alternatively, set the '{1}' property to a non-null value in the '{2}' constructor.</value>
+  </data>
+  <data name="ComplexObjectModelBinder_NoSuitableConstructor_ForType" xml:space="preserve">
+    <value>Could not create an instance of type '{0}'. Model bound complex types must not be abstract or value types and must have a parameterless constructor. Record types must have a single primary constructor.</value>
+  </data>
+  <data name="ValidationStrategy_MappedPropertyNotFound" xml:space="preserve">
+    <value>No property found that maps to constructor parameter '{0}' for type '{1}'. Validation requires that each bound parameter must have a property to read the value.</value>
```

Comment:
Suggest mentioning record types in this message e.g. "each bound parameter of a record's primary constructor"